src/javascript/app/pages/cashier/cashier.js

```diff
@@ -152,13 +152,42 @@ const Cashier = (() => {
         });
     };
 
+    const setBtnDisable = selector => $(selector).addClass('button-disabled').click(false);
+    
+    const applyStateLockLogic = (status, deposit, withdraw) => {
+        const cashier_lock = { is_on: status.includes('cashier_locked'), selectors: [deposit, withdraw] };
+        const withdrawal_lock = { is_on: status.includes('withdrawal_locked'), selectors: [withdraw] };
+        const no_withdrawal_or_trading = { is_on: status.includes('no_withdrawal_or_trading'), selectors: [withdraw] };
+        const unwelcome = { is_on: status.includes('unwelcome'), selectors: [deposit] };
+        if (cashier_lock.is_on) {
+            cashier_lock.selectors.forEach(selector => setBtnDisable(selector));
+        }
+        if (unwelcome.is_on) {
+            unwelcome.selectors.forEach(selector => setBtnDisable(selector));
+        }
+        if (withdrawal_lock.is_on || no_withdrawal_or_trading.is_on) {
+            withdrawal_lock.selectors.forEach(selector => setBtnDisable(selector));
+        }
```

Comment:
instead of checking individually by `if` condition, we can add a statuses_to_check array like 
```
const statuses_to_check = ['cashier_lock', 'withdrawal_lock', 'no_withdrawal_or_trading', 'unwelcome'];
```
then loop statuses_to_check and if any of them has is_on then for each selector disable the button.

I think that makes the code more readable and makes use of the unified object structure.

Reply:
and yea more functional :+1: